langserver/internal/refs/refs.go

```diff
@@ -0,0 +1,337 @@
+package refs
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"log"
+	"strconv"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+type Def struct {
+	// ImportPath is the import path at which the definition is located.
+	ImportPath string
+
+	// Path is the path of the definition (e.g. ["Router", "Route"] for a
+	// method named "Route" with receiver type "Router").
+	Path string
+}
+
+// Ref represents a reference to a definition.
+type Ref struct {
+	// Def is the definition being referenced.
+	Def Def
+
+	// Position is the position of the reference.
+	Position token.Position
+}
+
+func (r *Ref) String() string {
+	return fmt.Sprintf("&Ref{Def.ImportPath: %q, Def.Path: %q, Position: \"%s:%d:%d (offset %d)\"}", r.Def.ImportPath, r.Def.Path, r.Position.Filename, r.Position.Line, r.Position.Column, r.Position.Offset)
```

Comment:
Go uses the convention of `#1234` to denote offsets. It would be a bit less verbose and easier to scan IMO if you formatted this like `%s:%d:%d:#%d`.